### PR TITLE
Lloyd

### PR DESCRIPTION
**Pull request by tactnician**

### CODE REVIEW

File: assignments/arrays.js [hunk 1]
```diff
@@ -63,34 +63,131 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
+console.log(`The last car has id ${inventory[lastCar].id} and is a ${inventory[lastCar].car_year} ${inventory[lastCar].car_make} ${inventory[lastCar].car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
+
+// const sortAlpha = function(arr){
+//     for(let i=0; i<arr.length; i++){
+//         carModels.push(arr[i].car_make);
+//     };
+//     retrun carModels;
```
Comment (DaftBeowulf) [inline assignments/arrays.js:92]: I see you never actually invoke this, did you mean to leave these functions in? there's a typo on line 92 which would have stopped a return anyway, but in the future just delete unused/broken functions like this

Comment (tactnician) [reply assignments/arrays.js:92]: I tries several iterations of the sorting function. That one is depricated.. It needs to be removed

File: assignments/arrays.js [hunk 2]
```diff
@@ -63,34 +63,131 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
+        for(let j = 1; j<=i; j++){
+            if(arr[j-1].car_model>arr[j].car_model){
+                let temp = arr[j-1];
+                arr[j-1] = arr[j];
+                arr[j] = temp;
+            }
+        }
+    }
+    return arr;
+}
```
Comment (DaftBeowulf) [inline assignments/arrays.js:134]: I feel you were getting in the weeds a little bit here and over-complicating things with all the functions leading up to this as well as your implementation. if a function ends up not working or being unused you can just delete it to make your project easier to read. you started to go in the right direction with `.sort()`, all you had to do here was iterate over the inventory and push all the car models into an empty array, apply `.sort()` with no parameters which automatically sorts it alphabetically, and console log the new array

File: assignments/arrays.js [hunk 3]
```diff
@@ -63,34 +63,131 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
+const sortByAge = (arr, modelYear) => {
+    let arr2 = [];
+    for(let i = 0; i < arr.length; i++){
+        if(modelYear > arr[i].car_year){
+            arr2.push(arr[i]);
+        }
+    }
+    return arr2;
+}
+oldCars = sortByAge(inventory, 2000);
```
Comment (DaftBeowulf) [inline assignments/arrays.js:170]: this works but read the instruction again carefully, you already pushed all the car years into carYears array on line 154 and you were supposed to perform this challenge on that (would also save you a little time)

File: assignments/arrays.js [hunk 4]
```diff
@@ -63,34 +63,131 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
...
+            arr2.push(arr[i]);
+        } else if(arr[i].car_make.toLowerCase() === make2.toLowerCase()) {
+            arr2.push(arr[i]);
+        }
+    }
+    return arr2;
+}
+
+BMWAndAudi = makeSearch(inventory, 'BMW', "AUDI");
+console.log(BMWAndAudi);
```
Comment (DaftBeowulf) [inline assignments/arrays.js:190]: well done and good job making use of `.toLowerCase()` to account for an edge case of different case conventions, but remember you need to `JSON.stringify()` the final result per the instructions

File: assignments/objects.js [hunk 5]
```diff
@@ -4,11 +4,47 @@
...
 // 5,adaine5@samsung.com,Antonietta,F
-
+const Antonietta = {
+  "id": 5,
+  "name": ",Antonietta",
+  "email": "adaine5@samsung.com",
+  "gender": "F",
+  "multiplyNums" : (num1, num2) => {
+    return num1 * num2; 
+  }
```
Comment (DaftBeowulf) [inline assignments/objects.js:46]: good job, but remember with the ES6 syntax fat arrow functions you have a access to an implicit return: 
```"multiplyNums" : (num1, num2) => num1 * num2;```
does the same thing without having to be explciit and saves you a couple lines

File: assignments/arrays.js (hunk 2 above)
Comment (tactnician) [reply assignments/arrays.js:134]: After 20min of playing with sort I went back to my playbook and used a bubble sort that I found online a while back.

File: assignments/stretch-function-conversion.js [hunk 6]
```diff
@@ -1,23 +1,29 @@
 // Take the commented ES5 syntax and convert it to ES6 arrow Syntax
 
 // let myFunction = function () {};
+const myFunction = () => {};
 
 // let anotherFunction = function (param) {
 //   return param;
 // };
+const anotherFunction = (param) => {return param;}
```
Comment (DaftBeowulf) [inline assignments/stretch-function-conversion.js:9]: not incorrect, but you're not using all the advantages the fat arrow return offers. if you have only one parameter you can leave out the parentheses around it, you can leave out the curly braces around the actual function so long as it fits on one line, and with the implicit return you can actually leave that statement out. so this line would become 
```const anotherFunction = param => param```
which might look strange at first but will save you a  lot of time once you get the syntax down

**Review by DaftBeowulf — COMMENTED** [review]
I'm glad to see you're able to handle for loops well Lloyd, but remember to be minimalistic when you can in your code and delete commented out, unused functions, I like seeing the different approaches you came up with but it'd make it a lot easier to read. read up a little more on ES6 syntax particularly relating to the `=>` returns as your syntax in the assignment works but you're writing more code than necessary

File: assignments/arrays.js (hunk 2 above)
Comment (DaftBeowulf) [reply assignments/arrays.js:134]: bubble sorts can be crazy and I'm glad to see you pushing to use something more complex like that, I just wanted you to be aware that there was a simpler solution :+1:

**Pull request merged**